src/crimson/common/config_proxy.h

```diff
@@ -69,36 +70,38 @@ class ConfigProxy : public seastar::peering_sharded_service<ConfigProxy>
                                                const std::string &key) {
                                       rev_obs[obs].insert(key);
                                     }, nullptr);
-      for (auto& [obs, keys] : rev_obs) {
-        (*obs)->handle_conf_change(owner, keys);
-      }
-
-      return seastar::parallel_for_each(std::views::iota(1u, seastar::smp::count),
-                                        [&owner, new_values] (auto cpu) {
-        return owner.container().invoke_on(cpu,
-          [foreign_values = seastar::make_foreign(new_values)](ConfigProxy& proxy) mutable {
-            proxy.values.reset();
-            proxy.values = std::move(foreign_values);
-
-            std::map<std::string, bool> changes_present;
-            for (const auto& change : proxy.values->changed) {
-              std::string dummy;
-              changes_present[change] = proxy.get_val(change, &dummy);
-            }
-
-            ObserverMgr<ConfigObserver>::rev_obs_map rev_obs;
-            proxy.obs_mgr.for_each_change(changes_present,
-              [&rev_obs](auto obs, const std::string& key) {
-                rev_obs[obs].insert(key);
-              }, nullptr);
-            for (auto& [obs, keys] : rev_obs) {
-              (*obs)->handle_conf_change(proxy, keys);
-            }
-          });
-        }).finally([new_values] {
-          new_values->changed.clear();
-        });
-      });
+      return seastar::async([rev_obs = std::move(rev_obs), &owner]() mutable {
```

Comment:
Ack, this seems to work well. However, please explain why do we need a seastar::thread here. This is not easy to understand without context.
Please mention that the **common** `mclock_handle_conf_change` needs a way to run async code to run mon client commands.
I think that it would be best to include this bit in a separated commit with a proper commit message. The logic added here is not trivial to follow.